docs/articles/documentation/test-api/obtaining-data-from-the-client.md

```diff
@@ -45,6 +46,22 @@ import { ClientFunction } from 'testcafe';
 const getWindowLocation = ClientFunction(() => window.location);
 ```
 
+### Running Asynchronous Client Code
+
+You can create client functions that run asynchronous code.
+To this end, pass a function that returns a Promise to the `ClientFunction` constructor.
+In this instance, the client function will complete only when this Promise resolves.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+сonst performAsyncOperation = ClientFunction(() => {
+    return Promise(resolve => {
+        window.setTimeout(resolve, 500); // some async operations
```

Comment:
lets use some simple async function e.g. `fs.writeFile` or something

Reply:
`fs.writeFile` on the client?

Reply:
oops, you right